src/crypto/OlmDevice.js

```diff
@@ -137,16 +141,29 @@ OlmDevice.prototype.init = async function() {
 };
 
 
-function _initialise_account(sessionStore, pickleKey, account) {
-    const e2eAccount = sessionStore.getEndToEndAccount();
-    if (e2eAccount !== null) {
-        account.unpickle(pickleKey, e2eAccount);
-        return;
-    }
+async function _initialise_account(sessionStore, cryptoStore, pickleKey, account) {
+    let removeFromSessionStore = false;
+    await cryptoStore.endToEndAccountTransaction((accountData, save) => {
```

Comment:
can we s/accountData/pickledAccount/ or somesuch? I'm finding myself a bit confused.